src/app/(main)/query-runner/_components/cql-editor.tsx

```diff
@@ -96,9 +96,17 @@ export function CqlEditor() {
 			fetchSizeRef.current === "unset" || !fetchSizeRef.current
 				? undefined
 				: Number.parseInt(fetchSizeRef.current);
+
+		const { username, password, ...rest } = currentConnectionRef.current;
+		const connection = {
+			...rest,
+			username: username ?? null,
+			password: password ?? null,
```

Comment:
```suggestion
			username,
			password
```

This `Nullish coalescing` being used this way doesn't make much sense, if it's null you'll be attributing null to it again.

Reply:
> This `Nullish coalescing` being used this way doesn't make much sense, if it's null you'll be attributing null to it again.

You're right, using nullish coalescing like that didn't make sense. I removed it and instead added auth conditionally, only if both username and password are present. This way, if either is null, auth won't be included at all.